audio/drivers/alsa.c

```diff
@@ -20,160 +20,79 @@
 #include <string/stdstring.h>
```

Comment:
This file originally contained microphone support before I split it into a separate driver. Along the way, I commented code I didn't understand so that I could be ready to modify it. I also refactored parts of this driver to share code with the new mic driver.